dirsrvtests/tests/suites/schema/schema_csn_replication_test.py

```diff
@@ -0,0 +1,386 @@
+    user2 = users2.get(test_user.get_attr_val_utf8('uid'))
+    assert user2.get_attr_val_utf8('description') == test_user.get_attr_val_utf8('description')
+
+    log.info('Check schema CSN synchronization with longer wait if needed')
+    wait_for_schema_csn_sync(supplier1, supplier2)
+
+
+def test_schema_learning_with_paused_agreement(topology_m2, schema_csn_setup, request):
+    """Test schema learning when replication agreement is paused
+
+    :id: 9c8d7b6a-5e4f-3a2b-1c0d-8e7d6c5b4a38
+    :setup: Two supplier replication topology
+    :steps:
+        1. Pause replication agreement from supplier2 to supplier1
+        2. Add new schema elements to supplier2
+        3. Trigger replication from supplier1 to supplier2 (one direction only)
+        4. Verify schema CSNs show expected difference
+    :expectedresults:
+        1. Agreement is paused successfully
+        2. Schema is added to supplier2
+        3. Replication works in one direction
+        4. Schema CSNs show supplier2 has newer schema than supplier1
+    """
+    supplier1 = topology_m2.ms["supplier1"]
+    supplier2 = topology_m2.ms["supplier2"]
+    test_users = schema_csn_setup
+    test_user = test_users[2]
+    repl = ReplicationManager(DEFAULT_SUFFIX)
+
+    log.info('Testing schema learning with paused agreement')
+
+    replica2 = Replicas(supplier2).get(DEFAULT_SUFFIX)
+    agreements2 = Agreements(supplier2, replica2.dn)
+    agmt_list = agreements2.list()
+    assert len(agmt_list) >= 1
+
+    agmt_to_pause = agmt_list[0]
+    agmt_to_pause.pause()
+    request.addfinalizer(agmt_to_pause.resume)
+    log.info(f"Paused agreement: {agmt_to_pause.dn}")
+
+    schema2 = Schema(supplier2)
+
+    custom_at3 = create_custom_attributetype('testAttr', 3)
+    schema2.add_attributetype(custom_at3)
+    log.info(f"Added custom attribute type to supplier2: {custom_at3['names'][0]}")
+
+    custom_oc3 = create_custom_objectclass('testClass', 3)
+    schema2.add_objectclass(custom_oc3)
+    log.info(f"Added custom object class to supplier2: {custom_oc3['names'][0]}")
+
+    trigger_replication(repl, supplier1, supplier2, test_user)
+
+    users2 = UserAccounts(supplier2, DEFAULT_SUFFIX)
+    user2 = users2.get(test_user.get_attr_val_utf8('uid'))
+    assert user2.get_attr_val_utf8('description') == test_user.get_attr_val_utf8('description')
+
+    log.info('Verify schema CSNs are different (replication was paused)')
+    schema1 = Schema(supplier1)
+    schema2 = Schema(supplier2)
+
+    schema_csn_supplier1 = schema1.get_schema_csn()
+    schema_csn_supplier2 = schema2.get_schema_csn()
+
+    assert schema_csn_supplier1 is not None, "Supplier1 schema CSN should not be None"
+    assert schema_csn_supplier2 is not None, "Supplier2 schema CSN should not be None"
+    assert schema_csn_supplier1 != schema_csn_supplier2
+
+
+def test_schema_csn_sync_after_agreement_resume(topology_m2, schema_csn_setup):
+    """Test schema CSN synchronization after resuming paused agreement
```

Comment:
I do not see where the RA pause. For me replication occurs as soon as test_user is updated